### DISCUSSION TITLE
Changing training domain mid training

### DISCUSSION THREAD

**Discussion by nirmal129** [lululxvi/deepxde#1651] [Q&A]
Hello, 

I want to construct a model with two different training domains, $(t, x)\in[0, 25]\times[-50, 150]$ and $(t, x)\in[0, 50]\times[-50, 150]$. I want to train the model in such a way that it trains on the first domain for 5000 ADAM iterations and then L-BFGS, then it trains on the second domain (without reinitializing the weights) for 10000 ADAM iterations and then L-BFGS. Is it possible? If so, could someone provide me with a brief code snippet? 

Thank you!

**EDIT**

I looked up the code and realized/speculate that the weights of nn are separated from the model instance. Using this, I have the following code

```python3
# 1st training

model = dde.Model(data1, net)

model.compile("adam", lr=1e-3, loss_weights=loss_weights)
loss_history, train_state = model.train(iterations=num_iter, callbacks=[pde_residual_resampler])

model.compile("L-BFGS", loss_weights=loss_weights)
loss_history, train_state = model.train(callbacks=[pde_residual_resampler])

# 2nd training

model = dde.Model(data2, net)

model.compile("adam", lr=1e-3, loss_weights=loss_weights)
loss_history, train_state = model.train(iterations=num_iter, callbacks=[pde_residual_resampler])

model.compile("L-BFGS", loss_weights=loss_weights)
loss_history, train_state = model.train(callbacks=[pde_residual_resampler])
```

where data1 and data2 are the above mentioned domains. Am I on the correct path?

**Comment by lululxvi** [comment]
In your code, you have two independent models.

You can train the first set of points, stop the code and save the model, change the points, load the previous model, and then continue training.

**Reply by nirmal129** [reply]
Thanks for the response! 

Btw what do you mean by stop the code? Can't I do something like the following

```python3
model = dde.Model(initData, net)
model.compile()
model.train()
model.save()

net = newNet()
model = dde.Model(newData, net)
model.compile()
model.restore()
```

**Reply by lululxvi** [reply]
Your code may or may not work. I mean stoping the program.

Check the example of save and load models.
